spec/acceptance/redis_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'redis' do
+  password = ENV.fetch('ICINGA_REPO_PASSWORD', nil)
+  user = ENV.fetch('ICINGA_REPO_USER', nil)
+  # TODO: the secrets are only required for EL tests, so we can limit the scope further down
+  it 'works idempotently with no errors', if: password && user do
+    pp = <<~MANIFEST
+      if $facts['os']['family'] == 'RedHat' and $facts['os']['name'] != 'Fedora' {
+        yumrepo { 'icinga-stable-release':
+          descr    => 'ICINGA (stable release for epel)',
+          baseurl  => 'https://packages.icinga.com/subscription/rhel/$releasever/release/',
+          enabled  => 1,
+          gpgcheck => 1,
+          gpgkey   => 'https://packages.icinga.com/icinga.key',
+          username => "#{user}",
+          password => "#{password}",
```

Comment:
I suspect they will be logged in plain text which doesn't sound like a good idea. Perhaps you can somehow set them via Hiera?

Reply:
I won't be logged: https://github.com/puppetlabs/puppetlabs-yumrepo_core/blob/main/lib/puppet/type/yumrepo.rb#L458

puppet agent will just say `[redacted]`